### PR TITLE
cc_growpart.py: fix handling of /dev/mapper devices for Alpine

### PR DESCRIPTION
**Pull request by dermotbradley**
## Proposed Commit Message

```
feat(Alpine) cc_growpart.py: fix handling of /dev/mapper devices for Alpine

Unlike the majority of other Linux distros /dev/mapper entries in
Alpine Linux are NOT symlinks to /dev/dm-X block devices but rather are
actual block devices themselves. This is a side-effect of Alpine's use
of mdev, rather than udev, in its initramfs.

Therefore this change works out the corresponding /dev/dm-X for any
/dev/mapper/XYZ device on Alpine.
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [x] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [x] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by holmanb — CHANGES_REQUESTED** [review]
Thanks for putting this together @dermotbradley!

I have one minor change request for using deventry.path rather than manually creating that value using deventry.name.

**Comment by holmanb** [comment]
@dermotbradley This change reveals that the previous codepath assumed linux / gnu / systemd. For a better abstraction, `get_mapped_device()` probably deserves to be a static method on our Distro base class with your changes in distros/alpine.py, but I see that we have some freebsd-specific code in this module that should also make this transition so I'm happy to refactor both this and the FreeBSD code in a followup PR if you'd rather not do that in this PR - just let me know which you'd prefer.

File: cloudinit/config/cc_growpart.py [hunk 1]
```diff
@@ -368,6 +368,32 @@ def get_mapped_device(blockdev):
...
+
+            # Get the major/minor of the /dev/mapper block device
+            major = os.major(os.stat(blockdev).st_rdev)
+            minor = os.minor(os.stat(blockdev).st_rdev)
+
+            # Find the /dev/dm-X device with the same major/minor
+            with os.scandir("/dev/") as it:
+                for deventry in it:
+                    if deventry.name.startswith("dm-"):
+                        fulldevname = "/dev/" + deventry.name
```
[the remark this answers is not in the document]
Comment (dermotbradley) [reply cloudinit/config/cc_growpart.py]: Ok, changed now.

File: cloudinit/config/cc_growpart.py [hunk 2]
```diff
@@ -358,7 +358,7 @@ def devent2dev(devent):
     return dev
 
 
-def get_mapped_device(blockdev):
+def get_mapped_device(blockdev, distro_name):
```
[the remark this answers is not in the document]
Comment (dermotbradley) [reply cloudinit/config/cc_growpart.py:361]: As you mentioned below refactoring this in a separate PR along with the FreeBSD-related stuff would make sense.

**Review by holmanb — APPROVED** [review]
LGTM, thanks @dermotbradley!

**Pull request merged**